examples/pytorch/rgcn/entity_sample.py

```diff
@@ -0,0 +1,199 @@
+"""
+Differences compared to tkipf/relation-gcn
+* l2norm applied to all weights
+* remove nodes that won't be touched
+"""
+import argparse
+import torch as th
+import torch.nn.functional as F
+import dgl
+
+from dgl.dataloading import MultiLayerNeighborSampler, NodeDataLoader
+from torchmetrics.functional import accuracy
+from tqdm import tqdm
+
+from entity_utils import load_data
+from model import RelGraphEmbedLayer, RGCN
+
+def init_dataloaders(args, g, train_idx, test_idx, target_idx, device, use_ddp=False):
+    fanouts = [int(fanout) for fanout in args.fanout.split(',')]
+    sampler = MultiLayerNeighborSampler(fanouts)
+
+    train_loader = NodeDataLoader(
+        g,
+        target_idx[train_idx],
+        sampler,
+        use_ddp=use_ddp,
+        device=device,
+        batch_size=args.batch_size,
+        shuffle=True,
+        drop_last=False)
+
+    # The datasets do not have a validation subset, use the train subset
+    val_loader = NodeDataLoader(
+        g,
+        target_idx[train_idx],
+        sampler,
+        use_ddp=use_ddp,
+        device=device,
+        batch_size=args.batch_size,
+        shuffle=False,
+        drop_last=False)
+
+    test_sampler = MultiLayerNeighborSampler([None] * len(fanouts))
+    test_loader = NodeDataLoader(
+        g,
+        target_idx[test_idx],
+        test_sampler,
+        use_ddp=use_ddp,
+        device=device,
+        batch_size=32,
+        shuffle=False,
+        drop_last=False)
+
+    return train_loader, val_loader, test_loader
+
+def init_models(args, device, num_nodes, num_classes, num_rels):
+    embed_layer = RelGraphEmbedLayer(device,
+                                     num_nodes,
+                                     args.n_hidden,
+                                     dgl_sparse=args.dgl_sparse)
+
+    model = RGCN(args.n_hidden,
+                 args.n_hidden,
+                 num_classes,
+                 num_rels,
+                 num_bases=args.n_bases,
+                 dropout=args.dropout,
+                 self_loop=args.use_self_loop)
+
+    return embed_layer, model
+
+def process_batch(inv_target, batch):
+    _, seeds, blocks = batch
+    # map the seed nodes back to their type-specific ids,
+    # in order to get the target node labels
+    seeds = inv_target[seeds]
+
+    for blc in blocks:
+        _, v, eid = blc.edges(form='all')
+        _, inverse_index, count = th.unique(v, return_inverse=True, return_counts=True)
+        degrees = count[inverse_index]
+        norm = th.ones(eid.shape[0]).to(degrees.device) / degrees
+        blc.edata['norm'] = norm.unsqueeze(1)
```

Comment:
Is this normalizer calculation a common thing? Shall we add it as a DGL utility?

Reply:
Added `norm_by_dst` in `transform.py`.